hbase-client/src/test/java/org/apache/hadoop/hbase/util/TestThreadLocalPoolMap.java

```diff
@@ -43,42 +43,42 @@ protected PoolType getPoolType() {
   }
 
   @Test
-  public void testSingleThreadedClient() throws InterruptedException, ExecutionException {
+  public void testSingleThreadedClient() throws InterruptedException {
     Random rand = ThreadLocalRandom.current();
-    String randomKey = String.valueOf(rand.nextInt());
-    String randomValue = String.valueOf(rand.nextInt());
-    // As long as the pool is not full, we should get back what we put
-    runThread(randomKey, randomValue, randomValue);
-    assertEquals(1, poolMap.size(randomKey));
+    String key = "key";
+    String value = "value";
+    runThread(key, () -> value, value);
+    assertEquals(1, poolMap.values().size());
   }
 
   @Test
-  public void testMultiThreadedClients() throws InterruptedException, ExecutionException {
-    Random rand = ThreadLocalRandom.current();
-    // As long as the pool is not full, we should get back what we put
-    for (int i = 0; i < POOL_SIZE; i++) {
-      String randomKey = String.valueOf(rand.nextInt());
-      String randomValue = String.valueOf(rand.nextInt());
-      runThread(randomKey, randomValue, randomValue);
-      assertEquals(1, poolMap.size(randomKey));
+  public void testMultiThreadedClients() throws InterruptedException {
+    for (int i = 0; i < KEY_COUNT; i++) {
+      String key = Integer.toString(i);
+      String value = Integer.toString(2 * i);
+      runThread(key, () -> value, value);
     }
-    String randomKey = String.valueOf(rand.nextInt());
+
+    assertEquals(KEY_COUNT, poolMap.values().size());
+    poolMap.clear();
+
+    String key = "key";
     for (int i = 0; i < POOL_SIZE; i++) {
-      String randomValue = String.valueOf(rand.nextInt());
-      runThread(randomKey, randomValue, randomValue);
-      assertEquals(i + 1, poolMap.size(randomKey));
+      String value = Integer.toString(i);
+      runThread(key, () -> value, value);
     }
+
+    assertEquals(POOL_SIZE, poolMap.values().size());
   }
 
   @Test
-  public void testPoolCap() throws InterruptedException, ExecutionException {
-    Random rand = ThreadLocalRandom.current();
-    String randomKey = String.valueOf(rand.nextInt());
+  public void testPoolCap() throws InterruptedException {
```

Comment:
nit: a comment in the javadoc for this test that says "There is no pool cap for the ThreadLocalPool" or a change in the test name to e.g. `testNoPoolCap` would make this test easier to follow.